EDU-17928 - Setting up extensions for VTEX Sales App (#2672)
vtexdocs/dev-portal-content: docs/guides/VTEX-Sales-App/setting-up-extensions-for-vtex-sales-app.md

---
title: "Setting up extensions for VTEX Sales App"
slug: "setting-up-extensions-for-vtex-sales-app"
excerpt: "Learn how to set up extensions for VTEX Sales App in your store."
hidden: false
createdAt: "2026-05-26T15:59:53.407Z"
seeAlso:
  - "/docs/guides/creating-a-vtex-sales-app-extension"
  - "/docs/guides/vtex-sales-app-extension-points"
  - "/docs/guides/defineextensions-function"
---

> ℹ️ This feature is currently in beta, which means we're working to improve it. If you have any questions, please contact our [Support](https://help.vtex.com/pt/support).

[VTEX Sales App Extensibility](https://help.vtex.com/docs/tutorials/extensibility-in-vtex-sales-app) lets you customize the default assisted-sales journey with features that support your business requirements.

By using predefined extension points, you can integrate external APIs, use VTEX data, and add custom experiences to areas such as cart and checkout, the side menu, and the product details page.

> In this guide, you'll learn how to set up extension points for the VTEX Sales App of your store.

Author: ```suggestion
In this guide, you'll learn how to set up extensions for the VTEX Sales App of your store.
```

> In this guide, you'll learn how to set up extension points for VTEX Sales App in your FastStore project.

Author: ```suggestion
In this guide, you'll learn how to set up extension points for the VTEX Sales App of your store.
```

* The store account name for which you're creating extensions.
* The module you want to initialize a project. In this case, it's `sales-app`.
* The path where you want to initialize the project.

Run the following commands in your terminal:

```yarn
yarn fsp create
```

```npx
npx fsp create
```

As an example, consider the account `store-a` and the path `./store-a/sales-app`. Remember to fill in the prompts according to your project requirements:

```text
> ? What's the account name? store-a
> ? Which module do you want to initialize? sales-app
> ? What should be the path to initialize sales-app? ./store-a/sales-app
```

After running the command, the folder is created at the root of your monorepo based on the path you provided. Your monorepo structure should look similar to this:

```txt
📂 account-monorepo
│
├──📂 store-a
│   └──📂 sales-app
│      └──📂 src
│         └── 📂 components
│         │  └──📄 Example.tsx
│         ├──📄 Example.css
│         ├──📄 index.tsx
│         └──📄 package.json
└──📄faststore.json
```

#### Project structure

The `src/index.tsx` file at the root of your project is the entry point for all VTEX Sales App extensions you create. It looks like this:

> ```tsx

Author: ```suggestion
```typescript src/index.tsx
```

import { defineExtensions } from '@vtex/sales-app'
import { Example } from './components/Example'

export default defineExtensions({
  'cart.cart-list.after': Example,
})
```

> The `defineExtensions` function connects the extensions you create to the extension point where they should appear. You can create your extensions in separate files and then import them into `index.tsx` to keep your project organized.

Author: ```suggestion
The [`defineExtensions`](https://developers.vtex.com/docs/guides/defineextensions-function) function connects the extensions you create to the extension point where they should appear. You can create your extensions in separate files and then import them into `index.tsx` to keep your project organized.
```

> This function also helps during development by showing available extension points with autocomplete and type-checking, making the setup easier and less error-prone. To build your extension, use the hooks, types, and helper functions available in the `@vtex/sales-app` package. For more information, see the [VTEX Sales App extension points](#LINK) guide.

Author: https://github.com/vtexdocs/dev-portal-content/pull/2673
